### ISSUE TITLE
launcher setting ignored in preferences.txt

### ISSUE THREAD

**Issue by hamoid** [benfry/processing4#247]
## Description

Executing "Open Sketch Folder" on Arch Linux with i3wm and fish does not open the folder. I used to work around this by specifying a custom `launcher` setting. This worked in Processing 3.4, but not in later versions.

This affects very few people I guess (maybe 1?) but it would be nice if the launcher setting was enforced if present, as an override. The related lines are found here:

https://github.com/processing/processing4/blob/master/app/src/processing/app/platform/LinuxPlatform.java#L118

## Expected Behavior

I can use Open Sketch Folder

## Current Behavior

I can't use Open Sketch Folder. It opens a terminal window with text being printed forever in a loop.

## Steps to Reproduce

Press Ctrl+K

## Your Environment

* Processing version: 4.0b
* Operating System and OS version: up to date Arch Linux
* Other information: using i3 window manager

## Possible Causes / Solutions

I think a solution might be to tweak the order of the if statements: if a launcher is specified, use it, otherwise, try use the default approaches. But I don't know how messy touching any of that might be.

Old post about this issue:
https://discourse.processing.org/t/launcher-seems-to-be-ignored-in-the-preferences/10840/7

**Comment by benfry** [comment]
Yeah, I changed it to use Java's `Desktop` class instead so that we could avoid more custom code like this. (Meaning that your issue is really with Java, and how it implements `Desktop`…) 

Maybe I could change it to see if you have a launcher setting specified, but that also means changing up some of that other logic, rewriting `openFolderAvailable()` so that it doesn't set `launcher`. Unfortunately, that also likely means that some people will be disappointed if they accidentally had a `launcher` value set and wanted the `Desktop` behavior. Will think about it.

**Comment by hamoid** [comment]
I was concerned that it might not be so easy, that's why I didn't mention it earlier.
If no one else is affected it's very ok to close this.
